cli/src/cluster_query.rs

```diff
@@ -917,14 +912,29 @@ pub fn process_ping(
     })
     .expect("Error setting Ctrl-C handler");
 
-    let mut last_blockhash = Hash::default();
     let mut submit_count = 0;
     let mut confirmed_count = 0;
     let mut confirmation_time: VecDeque<u64> = VecDeque::with_capacity(1024);
 
+    let (mut blockhash, mut fee_calculator) = rpc_client.get_recent_blockhash()?;
+    let mut blockhash_transaction_count = 0;
+    let mut blockhash_acquired = Instant::now();
     'mainloop: for seq in 0..count.unwrap_or(std::u64::MAX) {
-        let (recent_blockhash, fee_calculator) = rpc_client.get_new_blockhash(&last_blockhash)?;
-        last_blockhash = recent_blockhash;
+        let now = Instant::now();
+        if now.duration_since(blockhash_acquired).as_secs() > 60 {
```

Comment:
@CriesofCarrots - this is where https://github.com/solana-labs/solana/pull/10162 will be useful!